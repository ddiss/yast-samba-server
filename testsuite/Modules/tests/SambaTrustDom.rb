# encoding: utf-8

module Yast
  class SambaTrustDomClient < Client
    def main
      #testedfiles: SambaTrustDom.pm

      Yast.include self, "testsuite.rb"

      @READ = {}
      @WRITE = {}
      @EXEC = {}

      @EXEC1 = { "target" => { "bash" => 0 } }

      @EXEC2 = { "target" => { "bash" => 1 } }

      TESTSUITE_INIT([@READ, @WRITE, @EXEC], nil)

      Yast.import "SambaTrustDom"

      DUMP("------------------------------------------------------------")
      TEST(lambda { SambaTrustDom.Establish("domain", "password") }, [
        @READ,
        @WRITE,
        @EXEC1
      ], nil)
      TEST(lambda { SambaTrustDom.Establish("domain", "password") }, [
        @READ,
        @WRITE,
        @EXEC2
      ], nil)
      DUMP("------------------------------------------------------------")
      TEST(lambda { SambaTrustDom.Establish("domain", "password\"\"abc") }, [
        @READ,
        @WRITE,
        @EXEC1
      ], nil)
      TEST(lambda { SambaTrustDom.Establish("domain", "password\"\"abc") }, [
        @READ,
        @WRITE,
        @EXEC2
      ], nil)
      DUMP("------------------------------------------------------------")
      TEST(lambda { SambaTrustDom.Establish("domain", "pas''sword\"\"'\"abc") }, [
        @READ,
        @WRITE,
        @EXEC1
      ], nil)
      TEST(lambda { SambaTrustDom.Establish("domain", "pas''sword\"\"'\"abc") }, [
        @READ,
        @WRITE,
        @EXEC2
      ], nil)
      DUMP("------------------------------------------------------------")

      nil
    end
  end
end

Yast::SambaTrustDomClient.new.main
