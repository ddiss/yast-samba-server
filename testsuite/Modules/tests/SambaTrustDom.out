Dump	------------------------------------------------------------
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%password\"" 0
Return	true
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%password\"" 1
Return	nil
Dump	------------------------------------------------------------
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%password\\\"\\\"abc\"" 0
Return	true
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%password\\\"\\\"abc\"" 1
Return	nil
Dump	------------------------------------------------------------
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%pas''sword\\\"\\\"'\\\"abc\"" 0
Return	true
Execute	.target.bash "net rpc trustdom establish \"domain\" -U \"root%pas''sword\\\"\\\"'\\\"abc\"" 1
Return	nil
Dump	------------------------------------------------------------
